Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -207,6 +210,96 @@ theorem valuation_of_mk (f : Polynomial K) {g : Polynomial K} (hg : g ≠ 0) :
       (Polynomial.idealX K).intValuation f / (Polynomial.idealX K).intValuation g := by
   simp only [RatFunc.mk_eq_mk' _ hg, valuation_of_mk']
 
+end HeightOneSpectrum
+
+section TrivialOnConstants
+
+variable {Γ : Type*}
+
+section LinearOrderedCommMonoidWithZero
+
+variable [LinearOrderedCommMonoidWithZero Γ]
+
+section Algebra
+
+variable (L : Type*) [Field L] [Algebra K L] {v : Valuation L Γ}
+
+lemma valuation_aeval_monomial_eq_valuation_pow
+    (w : L) (n : ℕ) {a : K}
+    (hv : ∀ a : K, a ≠ 0 → v (algebraMap K L a) = 1)
+    (ha : a ≠ 0) : v ((monomial n a).aeval w) = (v w) ^ n := by
+  simp [← C_mul_X_pow_eq_monomial, map_mul, map_pow, one_mul, RatFunc.algebraMap_X, hv a ha]
+
+theorem valuation_aeval_eq_valuation_X_pow_natDegree_of_one_lt_valuation_X [Nontrivial Γ]
+    [PosMulStrictMono Γ] (w : L) (hpos : 1 < v w) (hv : ∀ a : K, a ≠ 0 → v (algebraMap K L a) = 1)
```

Comment:
Can't you make this `(hv : ∀ a : K, a ≠ 0 → v (algebraMap K L a) = 1)` assumption global (at least in this Algebra section, but actually you also keep on repeating it after you close it)?